server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java

```diff
@@ -90,17 +91,24 @@ protected void newResponseAsync(
                 TransportGetAllocationStatsAction.TYPE,
```

Comment:
Ah I just realised we need to run this action if the user requests either `allocations` or `fs` now, otherwise we won't get the full `FsInfo`. I think we're missing a test here, something like `rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/nodes.stats/80_allocation_stats.yml` but for the `fs` metric.

Reply:
Thanks for your efforts. Added rest `fs` metric test case.